### PR TITLE
feature: CircleCI new flow

### PR DESCRIPTION
**Pull request by iranreyes**
Mike nowak here, adding circleCI flow:

**Changes introduced**

- [x] Build-related changes

**Does this PR introduce a breaking change?** (check one)

Not a breaking change, but from this point on codeship is not supported out of the box.

**Did you test your solution?**

- Was tested within CircleCI

## Problem Description

Migrating from codeship to circleci. 

## Solution Description

Updates to a branch based workflow for circleCI. Try to ensure that you configure circleCI with "Only build pull requests" in Project Settings->Advanced since this is not necessary on _any_ commit (and likely costly.) 

Main Branch flow
Screen Shot 2022-07-14 at 15 27 56

PR Flow
Screen Shot 2022-07-14 at 15 30 15

I've put documentation around the place and also enabled auto-tagging. Auto-tagging might not be ideal because it automatically makes a new version on each main branch commit, but this automation should make some chore jobs easier.

Most of the tests are left empty by default, to be filled in future updates.

adviser-ci has been disabled for now in linters because it was failing on circleCI.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,604 @@
...
+#
+#################################
+workflows:
+  version: 2
+  pull-requests:
+    jobs:
+      # - envtest:
+      #     filters:
+      #       branches:
+      #         ignore: main
```
Comment (DonghyukJacobJang) [inline .circleci/config.yml:429]: should we delete this block too?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,604 @@
...
+      - deployment:
+          name: cache-invalidation-staging
+          requires:
+            - deployment-staging
+            # if you do add storybook to staging, make sure you invalidate cache after it's deployed
+            # - storybook-stage
+          ci_env: 'staging'
+          s3_origin_bucket: $S3_ORIGIN_DEV # change to _STAGE
+          cloudfront_distribution_id: $CLOUDFRONT_ID_DEV # change to _STAGE
+          aws_env_suffix: '_DEV' # change to _STAGE
```
Comment (DonghyukJacobJang) [inline .circleci/config.yml:554]: should we use a postfix `_STAGE` here?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,604 @@
...
+            - cache-invalidation-staging
+
+      - deployment:
+          name: deployment-production
+          requires:
+            - approve-build-production
+          ci_env: 'production'
+          s3_origin_bucket: $S3_ORIGIN_DEV # change to _PROD
+          cloudfront_distribution_id: $CLOUDFRONT_ID_DEV # change to _PROD
+          aws_env_suffix: '_DEV' # change to _PROD
```
Comment (DonghyukJacobJang) [inline .circleci/config.yml:581]: `_PROD` postfix?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,604 @@
...
+          script_name: 'deploy.sh'
+
+      - deployment:
+          name: cache-invalidation-prod
+          requires:
+            - deployment-production
+          ci_env: 'production'
+          s3_origin_bucket: $S3_ORIGIN_DEV # change to _DEV
+          cloudfront_distribution_id: $CLOUDFRONT_ID_DEV # change to _PROD
+          aws_env_suffix: '_DEV' # change to _PROD
```
Comment (DonghyukJacobJang) [inline .circleci/config.yml:591]: `_PROD` postfix?

**Comment by iranreyes** [comment]
@DonghyukJacobJang I will try to fill them this month, instead of just being "test". Let's just merge it and iterate from here

**Pull request merged**